src/redux/selectedThemeSlice.ts

```diff
@@ -0,0 +1,38 @@
+import { Theme, createTheme } from '@mui/material';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { themes } from '../utils/themes';
+
+const OSTheme = window
+  .matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+
+localStorage.setItem('OSTheme', OSTheme);
+
+const getTheme = (themeName: string): Theme => {
+  const theme = themes.find((foundTheme) => foundTheme.themeName === themeName)
+    || { themeName: '', themeObject: createTheme() };
+
+  return createTheme(theme.themeObject);
+};
```

Comment:
use object distructure:

```suggestion
const getTheme = (targetThemeName: string): Theme => {
  const theme = themes.find(({ themeName }) => themeName === targetThemeName)
    || { themeName: '', themeObject: createTheme() };

  return createTheme(theme.themeObject);
};
```